python/pyspark/sql/pandas/conversion.py

```diff
@@ -252,6 +242,74 @@ def _collect_as_arrow(self):
         # Re-order the batch list using the correct order
         return [batches[i] for i in batch_order]
 
+    def _collect_as_arrow_table(self, _force_split_batches=False):
```

Comment:
I don't think we really need to add this method, it just does the conversion from batches to DataFrame. The unit test doesn't even use the returned Table right?